src/qililab/calibration/calibration_node.py

```diff
@@ -0,0 +1,668 @@
+# Copyright 2023 Qilimanjaro Quantum Tech
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Automatic-calibration Node module, which works with notebooks as nodes."""
+import json
+import logging
+import os
+from datetime import datetime
+from io import StringIO
+from typing import Callable
+
+import numpy as np
+import papermill as pm
+
+from qililab.config import logger
+
+logger_output_start = "RAND_INT:47102512880765720413 - OUTPUTS: "
+
+
+class CalibrationNode:  # pylint: disable=too-many-instance-attributes
+    """Automatic calibration Node class representing a node in the calibration graph.
```

Comment:
Side note: there is WAY too many lines before the args are showed. Think that most of the users only care about the args.

All the "extra information" should be inside the `Examples` section right after the `Args`.

Reply:
Agree!  What would you think on having only this before the examples:
![Screenshot 2023-11-08 at 16 43 53](https://github.com/qilimanjaro-tech/qililab/assets/109400222/e7544878-4330-4fc5-b3e7-ecdfb4dee4c9)

And move the explanation of the run to examples? It's still long, but I think it makes sense to have that there! If you think otherwise tell me pls!

Reply:
Addressed!